ui/src/main/java/com/midtrans/sdk/ui/views/ewallet/xltunai/XlTunaiPresenter.java

```diff
@@ -0,0 +1,50 @@
+package com.midtrans.sdk.ui.views.ewallet.xltunai;
+
+import com.midtrans.sdk.core.MidtransCore;
+import com.midtrans.sdk.core.MidtransCoreCallback;
+import com.midtrans.sdk.core.models.snap.ewallet.xltunai.XlTunaiPaymentResponse;
+import com.midtrans.sdk.ui.MidtransUi;
+import com.midtrans.sdk.ui.abtracts.BasePaymentPresenter;
+import com.midtrans.sdk.ui.models.PaymentResult;
+
+/**
+ * Created by ziahaqi on 4/7/17.
+ */
+
+public class XlTunaiPresenter extends BasePaymentPresenter {
+
+    private final XLTunaiPaymentView view;
+    private PaymentResult<XlTunaiPaymentResponse> paymentResult;
```

Comment:
For more consistent data format, we can save the payment response instead.

Payment result will be generated after activity request it on the getter.